space_packet_parser/packets.py

```diff
@@ -2,34 +2,46 @@
 """Packet containers and parsing utilities for space packets."""
 
 from dataclasses import dataclass, field
-from typing import Union, Optional, Protocol, List
+from typing import Any, Optional, Protocol, List
 
 
-@dataclass
-class ParsedDataItem:
-    """Representation of a parsed parameter
+class DataItem:
+    """Mixin class for storing access to the raw value of a parsed data item.
 
-    Parameters
-    ----------
-    name : str
-        Parameter name
-    unit : str
-        Parameter units
-    raw_value : any
-        Raw representation of the parsed value. May be lots of different types but most often an integer
-    derived_value : float or str
-        May be a calibrated value or an enum lookup
-    short_description : str
-        Parameter short description
-    long_description : str
-        Parameter long description
+    We need to override the __new__ method to store the raw value of the data item
+    on immutable built-in types. So this is just a way of allowing us to inject our
+    own attribute into the built-in types.
     """
-    name: str
-    raw_value: Union[bytes, float, int, str]
-    unit: Optional[str] = None
-    derived_value: Optional[Union[float, str]] = None
-    short_description: Optional[str] = None
-    long_description: Optional[str] = None
+    def __new__(cls, value: Any, raw_value: Any = None) -> Any:
```

Comment:
Is this the signature we want on this class? I believe that every DataItem should have a raw_value, no matter what. The raw value should be the closest representation of the encoded data we can return.
e.g. `bytes` for strings and binary blobs. `int` for integers, etc. I think I want to write some brief documentation on the philosophy of these representations since there is no objective meaning of "raw" value. Again, not for this PR. I'll make an Issue.

Reply:
I added your text to the class docstring. I also made this a private class now to only be used internally.